freqtrade/tests/test_freqtradebot.py

```diff
@@ -201,27 +201,25 @@ def func(nb_assets=-1):
     assert result == -1
 
 
-def test_gen_pair_whitelist(mocker, default_conf, get_market_summaries_data) -> None:
+def test_gen_pair_whitelist(mocker, default_conf, tickers) -> None:
     """
     Test _gen_pair_whitelist() method
     """
     freqtrade = get_patched_freqtradebot(mocker, default_conf)
-    mocker.patch(
-        'freqtrade.freqtradebot.exchange.get_market_summaries',
-        return_value=get_market_summaries_data
-    )
+    mocker.patch('freqtrade.freqtradebot.exchange.get_tickers', tickers)
+    mocker.patch('freqtrade.freqtradebot.exchange.exchange_has', MagicMock(return_value=True))
 
     # Test to retrieved BTC sorted on BaseVolume
     whitelist = freqtrade._gen_pair_whitelist(base_currency='BTC')
-    assert whitelist == ['ZCL/BTC', 'ZEC/BTC', 'XZC/BTC', 'XWC/BTC']
+    assert whitelist == ['ETH/BTC', 'TKN/BTC', 'BLK/BTC']
```

Comment:
I see no reason to reduce the amount of pairs - if some are not supported on binance we should pick different ones (but don't reduce the count of pairs...)

Reply:
It was mostly so that the `tickers` fixture wouldn't be overly long. The exchange is mocked anyway so all pairs are accepted (again, like with the fees, so tests don't start failing just because the exchange is updated). But I can add another pair to the tickers fixture which I guess would make the test a bit more robust against randomly passing (1/6 -> 1/24 chance)

Reply:
Added a 4th pair https://github.com/gcarq/freqtrade/pull/603/commits/53b1f8d3a49709fe5d21aaaac4e719c5de049451